### PR TITLE
Should use secret driver in ACL for the encrypted volume creation

### PR DESCRIPTION
**Pull request by meinaLi**
Should use secret driver in ACL for the encrypted volume creation
Signed-off-by: Meina Li <meili@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_create.py [hunk 1]
```diff
@@ -279,6 +280,17 @@ def check_vol(pool_name, vol_name, expect_exist=True):
                 if params.get('setup_libvirt_polkit') == 'yes':
                     process.run("chmod 666 %s" % vol_xml, ignore_status=True,
                                 shell=True)
+                    if luks_encrypted and libvirt_version.version_compare(4, 5, 0):
+                        try:
+                            polkit = test_setup.LibvirtPolkitConfig(params)
+                            polkit_rules_path = polkit.polkit_rules_path
+                            cmd = "sed -i '/.secret./{n;s/QEMU/secret/g;s/pool_name/secret_uuid/;s/virt-dir-pool/%s/;}' %s" % (luks_secret_uuid, polkit_rules_path)
```
Comment (chloerh) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_create.py]: We should try to avoid using shell command lines in test scripts. It's better to use python lines instead.

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_create.py [hunk 2]
```diff
@@ -279,6 +280,17 @@ def check_vol(pool_name, vol_name, expect_exist=True):
...
+                            process.system(cmd, ignore_status=True)
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/volume/virsh_vol_create.py]: we should be backing up and revert the changes of polkit config once the test is done. please handle it as part of clean up.

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_create.py (hunk 1 above)
Comment (meinaLi) [reply libvirt/tests/src/virsh_cmd/volume/virsh_vol_create.py]: Finish to update.

File: libvirt/tests/src/virsh_cmd/volume/virsh_vol_create.py (hunk 2 above)
Comment (meinaLi) [reply libvirt/tests/src/virsh_cmd/volume/virsh_vol_create.py]: The polkit config file changed in script is newly created before test, and it will automatically clean up after test. So I think there's no need to back up it.

**Comment by chloerh** [comment]
Looks good to me
